Search/SentinelSearch.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * @function sentinelSearch
+ * @description Sentinel search algorithm for array.
+ *
+ * Sentinel linear search is a variation of the standard linear search algorithm used to
+ * find a target value in an array or list. The basic idea behind this algorithm is to add a
+ * sentinel value at the end of the array which is equal to the target value we are looking for.
+ * This helps to avoid checking the array boundary condition during each iteration of the loop,
+ * as the sentinel value acts as a stopper for the loop.
+ *
+ * @param {number[]} array - sorted list of numbers
+ * @param {number} target - target number to search for
+ * @return {number | null} - index of the target number in the list, or null if not found
+ * @see [SentinelSearch](https://www.geeksforgeeks.org/sentinel-linear-search/)
+ * @example sentinelSearch([1,2,3], 2) => 1
+ * @example sentinelSearch([4,5,6], 2) => null
+ * @complexity_analysis
+ * Time Complexity :
```

Comment:
I think just O(n) or omitting it altogether would suffice here. It's pretty trivial to see.